### PR TITLE
[TableGen][SelectionDAG] Make CheckValueTypeMatcher use MVT::SimpleValueType

### PR DESCRIPTION
**Pull request by 4vtomat**
The original `CheckValueTypeMatcher` stores StringRef as the member variable type, however it's more efficient to use use MVT::SimpleValueType since it prevents string comparison in isEqualImpl, it also reduce the memory consumption in each object.

### CODE REVIEW

**Review by arsenm — APPROVED** [review]

**Comment by 4vtomat** [comment]
Update commit message.

File: llvm/utils/TableGen/DAGISelMatcherGen.cpp [hunk 1]
```diff
@@ -235,7 +235,8 @@ void MatcherGen::EmitLeafMatchCode(const TreePatternNode &N) {
     if (N.hasName())
       return;
     // An unnamed ValueType as in (sext_inreg GPR:$foo, i8).
-    return AddMatcher(new CheckValueTypeMatcher(LeafRec->getName()));
+    return AddMatcher(new CheckValueTypeMatcher(
+        static_cast<MVT::SimpleValueType>(LeafRec->getValueAsInt("Value"))));
```
Comment (topperc) [inline llvm/utils/TableGen/DAGISelMatcherGen.cpp:239]: Use `llvm::getValueType(LeafRec)`

**Review by topperc — APPROVED** [review]
LGTM

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Pull request merged**